internal/pkg/instrumentation/bpf/net/http/client/probe.go

```diff
@@ -91,12 +92,25 @@ func New(logger logr.Logger) probe.Probe {
 				Key: "request_proto_pos",
 				Val: structfield.NewID("std", "net/http", "Request", "Proto"),
 			},
+			probe.StructFieldConst{
+				Key: "io_writer_buf_ptr_pos",
+				Val: structfield.NewID("std", "bufio", "Writer", "buf"),
+			},
+			probe.StructFieldConst{
+				Key: "io_writer_n_pos",
+				Val: structfield.NewID("std", "bufio", "Writer", "n"),
+			},
 		},
 		Uprobes: []probe.Uprobe[bpfObjects]{
 			{
 				Sym: "net/http.(*Transport).roundTrip",
 				Fn:  uprobeRoundTrip,
 			},
+			{
+				Sym:       "net/http.Header.writeSubset",
+				Fn:        uprobeWriteSubset,
+				DependsOn: []string{"net/http.(*Transport).roundTrip"},
```

Comment:
What is the goal of `DependsOn`?
Maybe we shouldn't load this uprobe if the kernel is in lockdown?

Reply:
I added a comment to clarify what this is about, but it's like @damemi  said. The writeSubset may appear in the binary of the program without roundTrip. That will force us to load the client probe and fail because roundTrip isn't optional, but also we'd be loading a bpf probe for no reason, where the dependent functionality isn't there. With DependsOn, we can specify that a uprobe should not be counted for activating the bpf program, unless the dependent symbol is found too.

Reply:
> Maybe we shouldn't load this uprobe if the kernel is in lockdown?

We discussed this at the SIG, there's some value of generating traces when we can't context propagate, like spanmetrics generation. Also customers might be confused why instrumentation doesn't work, a log message is easy to miss.